sdk/kas_client.go

```diff
@@ -287,26 +285,67 @@ func (k *KASClient) getRewrapRequest(keyAccess KeyAccess, policy string) (*kas.R
 	return &rewrapRequest, nil
 }
 
-type publicKeyWithID struct {
-	publicKey, kid string
+type kasKeyRequest struct {
+	url, algorithm string
+}
+
+type timeStampedKASInfo struct {
+	KASInfo
+	time.Time
+}
+
+// Caches the most recent key info for a given KAS URL and algorithm
+type kasKeyCache struct {
```

Comment:
We probably do want the application to have *some* control over it, although yes it may make sense to restrict access in some fashion. Notably, in the test code I have to call 'clear' after each test case that involves rotation. So I might want to change this.

Reply:
See https://github.com/opentdf/platform/issues/1107